### DISCUSSION TITLE
Custom badge from URL content

### DISCUSSION THREAD

**Discussion by svoop** [badges/shields#8809] [Q&A] [answered]
I'd like to add a schema version badge for our OFMX XML schema version, however, not based on the repo but rather the authoritative schema URI.

The current schema location is `http://schema.openflightmaps.org/0.1/OFMX-Snapshot.xsd`, however, it's hard for some of our tools to extract the version `0.1` from the URI, so I created a separate URL for this:

http://schema.openflightmaps.org/version.txt

It returns `0.1` as simple text/plain.

Is it possible somehow to create a version or custom badge which queries a URL and then uses the returned content in the badge?

Something along the lines of:

```
https://img.shields.io/text?label=schema&url=http://schema.openflightmaps.org/version.txt
```

To be rendered as:

![Schema](https://img.shields.io/badge/schema-0.1-blue)

**Comment by chris48s** [comment, marked as answer]
We've got 3 badges that can take an arbitrary url and extract a value from the content. We've got 2 that can extract a value from a JSON or yaml document using a jsonpath selector and one that can extract a value from an XML document using an xpath selector (although I'm not sure if xpath will work on a xsd schema).
We don't plan to add any additional formats, but if you could publish your version file in one of those formats you could use one of those. There is a builder for these on the front page https://shields.io/#dynamic-badge

**Reply by svoop** [reply]
Very nice, I was so focused on the different shield types, I didn't see the dynamic section further down. Thanks for the hint!
